packages/components/src/components/post-list/post-list.tsx

```diff
@@ -0,0 +1,64 @@
+import { Component, Element, Prop, Host, State, h } from '@stencil/core';
+import { version } from '@root/package.json';
+
+/**
+ * @slot default - Slot for placing the list title.
+ * @slot post-list-item - Slot for placing post-list-item components.
+ */
+
+@Component({
+  tag: 'post-list',
+  styleUrl: 'post-list.scss',
+  shadow: false,
+})
+export class PostList {
+  @Element() host: HTMLPostListElement;
+
+  /**
+   * The unique title of the list that is also referenced in the labelledby
+   */
+  @State() uuid: string;
+
+  /**
+   * If `true`, the list title will be hidden. Otherwise, it will be displayed.`
+   */
+  @Prop() readonly titleHidden: boolean = false;
+
+  /**
+   * The list can become horizontal by setting `horizontal="true"` or just `horizontal`
+   */
+  @Prop() readonly horizontal: boolean = false;
+
+  componentWillLoad() {
+    /**
+     * Get the id set on the host element or use a random id by default
+     */
+    this.uuid = `list-${crypto.randomUUID()}`;
+  }
+
+  componentDidLoad() {
+    const titleSlot = this.host.querySelector(`#${this.uuid}`).firstElementChild as HTMLSlotElement;
+    this.checkTitle(titleSlot);
+  }
+
+  private checkTitle(titleContent: HTMLSlotElement) {
+    if (!titleContent.innerText) {
+      throw new Error(
+        'Please provide a title to the list component. Title is mandatory for accessibility purposes.',
+      );
+    }
+  }
+
+  render() {
+    return (
+      <Host data-version={version}>
+        <div id={this.uuid} class={`list-title${this.titleHidden ? ' visually-hidden' : ''}`}>
```

Comment:
Instead of using a querySelector your could reference the element like so:
```suggestion
        <div ref={el => this.titleEl = el} class={`list-title${this.titleHidden ? ' visually-hidden' : ''}`}>
```